src/transactions/transaction_builder/helpers.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { EntryFunctionArgumentTypes, SimpleEntryFunctionArgumentTypes } from "../types";
+import { Bool, FixedBytes, MoveString, U128, U16, U256, U32, U64, U8 } from "../../bcs";
+import { AccountAddress } from "../../core";
+import { MoveFunction } from "../../types";
+
+export function isBool(arg: SimpleEntryFunctionArgumentTypes): arg is boolean {
+  return typeof arg === "boolean";
+}
+
+export function isString(arg: any): arg is string {
+  return typeof arg === "string";
+}
+
+export function isNumber(arg: SimpleEntryFunctionArgumentTypes): arg is number {
+  return typeof arg === "number";
+}
+
+export function isLargeNumber(arg: SimpleEntryFunctionArgumentTypes): arg is number | bigint | string {
+  return typeof arg === "number" || typeof arg === "bigint" || typeof arg === "string";
+}
+
+export function isNull(arg: SimpleEntryFunctionArgumentTypes): arg is null | undefined {
+  return arg === null || arg === undefined;
+}
+
+export function isBcsBool(arg: EntryFunctionArgumentTypes | SimpleEntryFunctionArgumentTypes): arg is Bool {
+  return arg instanceof Bool;
+}
+export function isBcsAddress(
+  arg: EntryFunctionArgumentTypes | SimpleEntryFunctionArgumentTypes,
+): arg is AccountAddress {
+  return arg instanceof AccountAddress;
+}
+export function isBcsString(arg: EntryFunctionArgumentTypes | SimpleEntryFunctionArgumentTypes): arg is MoveString {
+  return arg instanceof MoveString;
+}
+export function isBcsFixedBytes(arg: EntryFunctionArgumentTypes | SimpleEntryFunctionArgumentTypes): arg is FixedBytes {
+  return arg instanceof FixedBytes;
+}
+
+export function isBcsU8(arg: EntryFunctionArgumentTypes | SimpleEntryFunctionArgumentTypes): arg is U8 {
+  return arg instanceof U8;
+}
+export function isBcsU16(arg: EntryFunctionArgumentTypes | SimpleEntryFunctionArgumentTypes): arg is U16 {
+  return arg instanceof U16;
+}
+export function isBcsU32(arg: EntryFunctionArgumentTypes | SimpleEntryFunctionArgumentTypes): arg is U32 {
+  return arg instanceof U32;
+}
+export function isBcsU64(arg: EntryFunctionArgumentTypes | SimpleEntryFunctionArgumentTypes): arg is U64 {
+  return arg instanceof U64;
+}
+export function isBcsU128(arg: EntryFunctionArgumentTypes | SimpleEntryFunctionArgumentTypes): arg is U128 {
+  return arg instanceof U128;
+}
+export function isBcsU256(arg: EntryFunctionArgumentTypes | SimpleEntryFunctionArgumentTypes): arg is U256 {
+  return arg instanceof U256;
```

Comment:
I guess I could probably change all these input types to `any`

Reply:
I like strong typing